src/library/Uno.Material/Extensions/ControlExtensions.cs

```diff
@@ -222,6 +222,14 @@ public static class ControlExtensions
 
 		#endregion
 
+		private static void OnIconChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+		{
+			if (d is TextBox textBox)
```

Comment:
why constraint it to `TextBox` only? the target property accept any `Control` as recipient?

Reply:
But only TextBox supports LeadingIcon/Command/IsVisible, right?

In that case we should probably spit out a warning if someone is trying to set those properties on things other than a TextBox

Reply:
Should I use `System.Diagnostics.Debug.WriteLine("...")` for the warning?

Reply:
I used `d.Log().LogWarning("...")`

https://github.com/unoplatform/Uno.Themes/blob/fcbeafc893e87690299da55a23cca54224d5a995/src/library/Uno.Material/Extensions/ControlExtensions.cs#L247-L253